internal/test/e2e/fluxGit.go

```diff
@@ -0,0 +1,83 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/aws/eks-anywhere/internal/pkg/ssm"
+	"github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	e2etests "github.com/aws/eks-anywhere/test/framework"
+)
+
+type s3Files struct {
+	key, dstPath string
+	permission   int
+}
+
+func (e *E2ESession) setupFluxGitEnv(testRegex string) error {
+	re := regexp.MustCompile(`^.*GitFlux.*$`)
+	if !re.MatchString(testRegex) {
+		logger.V(2).Info("Not running Flux Generic Git Provider tests, skipping environment setup")
+		return nil
+	}
+
+	requiredEnvVars := e2etests.RequiredFluxGitEnvVars()
+	for _, eVar := range requiredEnvVars {
+		if val, ok := os.LookupEnv(eVar); ok {
+			e.testEnvVars[eVar] = val
+		}
+	}
+
+	for _, file := range buildFluxGitFiles(e.testEnvVars) {
+		if err := e.downloadFileInInstance(file); err != nil {
+			return fmt.Errorf("downloading flux-git file to instance: %v", err)
+		}
+	}
+
+	err := e.setUpSshAgent(e.testEnvVars[config.EksaGitPrivateKeyTokenEnv])
+	if err != nil {
+		return fmt.Errorf("setting up ssh agent on remote instance: %v", err)
+	}
+
+	return nil
+}
+
+func buildFluxGitFiles(envVars map[string]string) []s3Files {
+	return []s3Files{
+		{
+			key:        "git-flux/known_hosts",
```

Comment:
We can store these in secrets manager and load them that way instead of having to copy the file from S3 right? What are some other files we store on S3 that we copy over this way?

Reply:
Actually this is probably how we get `eksctl`?

Reply:
yep, this is how we get eksctl

the original pass at this did use the secrets manager approach and it was very ungainly; this is much easier and more clear, after discussion between @g-gaston and myselc.

the env vars are pointers to files, not values, so to use the env vars + secrets manager approach we need to not only pass the pointer value but also the contents, which gets pretty messey. this way, we easily copy the files from a known-good location and leave little room for error when interpolating the key value or knownhosts public keys into a file